### PR TITLE
[clang-tidy] Treat fields in anonymous records as names in enclosing scope when checking name styles

### PR DESCRIPTION
**Pull request by Lancern**
Currently, fields in anonymous records are treated as normal record members during naming style check. This can be undesirable in certain situations since these fields are used just like names in their enclosing scopes:

```c++
class Foo {
  union {
    int iv_;    // warning: invalid case style for public member 'iv_'
    float fv_;  // warning: invalid case style for public member 'fv_'
  };
};
```

`iv_` and `fv_` are used in the code like private members of `Foo` but their naming style comes from rules for public members.

This PR changes this behavior. It adds a new option `CheckAnonFieldInParent` to `readability-identifier-naming`. When set to `true`, fields in anonymous records will be treated as names in their enclosing scopes when checking name styles. Specifically:

- If the anonymous record is defined within the file scope or in a namespace scope, treat its fields as global variables when checking name styles;
- If the anonymous record is defined within a function, treat its fields as local variables when checking name styles;
- If the anonymous record is defined within a non-anonymous record, treat its fields as non-static record members when checking name styles.

### CODE REVIEW

**Comment by EugeneZelenko** [comment]
Please mention changes in Release Notes.

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -128,6 +128,11 @@ Improvements to clang-tidy
   as a value for `-export-fixes` to export individual yaml files for each
   compilation unit.
 
+- A new option `readability-identifier-naming.CheckAnonFieldInParent` is added. When set
```
Comment (EugeneZelenko) [inline clang-tools-extra/docs/ReleaseNotes.rst:131]: Please keep alphabetical order (by check name) in this section. If entry for `readability-identifier-naming` exists, append statement to it.

File: clang-tools-extra/docs/clang-tidy/checks/readability/identifier-naming.rst [hunk 2]
```diff
@@ -207,6 +208,31 @@ After if AggressiveDependentMemberLookup is `true`:
...
+
+    class Foo {
+    private:
+      union {
+        int iv_;
+        float fv_;
+      };
+    };
+
+If CheckAnonFieldInParent is `false`, you may get warnings that ``iv_`` and
```
Comment (PiotrZSL) [inline clang-tools-extra/docs/clang-tidy/checks/readability/identifier-naming.rst:230]: :option:`CheckAnonFieldInParent `

File: clang-tools-extra/docs/clang-tidy/checks/readability/identifier-naming.rst [hunk 3]
```diff
@@ -207,6 +208,31 @@ After if AggressiveDependentMemberLookup is `true`:
...
+``fv_`` are not coherent to public member names, because ``iv_`` and ``fv_``
+are public members of the anonymous union. When CheckAnonFieldInParent is
```
Comment (PiotrZSL) [inline clang-tools-extra/docs/clang-tidy/checks/readability/identifier-naming.rst:232]: :option:`CheckAnonFieldInParent`

**Review by PiotrZSL — APPROVED** [review]
LGTM

**Pull request merged**